core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -734,9 +740,9 @@ private[spark] class BlockManager(
    * @param keepReadLock if true, this method will hold the read lock when it returns (even if the
    *                     block already exists). If false, this method will hold no locks when it
    *                     returns.
-   * @return `Some(PutResult)` if the block did not exist and could not be successfully cached,
-   *         or None if the block already existed or was successfully stored (fully consuming
-   *         the input data / input iterator).
+   * @return `Some(Option[Iterator[Any])` if the block did not exist and could not be successfully
```

Comment:
This will be much easier to clean up if we split the monolithic `doPut()` into a `doPutBytes()` and `doPutIterator()`.